tensorbay/opendataset/RarePlanesSynthetic/loader.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+#
+# Copytright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+
+"""Dataloader of RarePlanesSynthetic dataset."""
+
+import os
+from itertools import takewhile
+from typing import Any, Callable, Dict, List, Tuple, Union
+
+import numpy as np
+
+from tensorbay.dataset import Data, Dataset
+from tensorbay.label import Classification, LabeledBox2D, PanopticMask
+from tensorbay.opendataset._utility import glob
+
+try:
+    from PIL import Image
+except ModuleNotFoundError:
+    from tensorbay.opendataset._utility.mocker import Image  # pylint:disable=ungrouped-imports
+try:
+    import xmltodict
+except ModuleNotFoundError:
+    from tensorbay.opendataset._utility.mocker import xmltodict  # pylint:disable=ungrouped-imports
+
+DATASET_NAME = "RarePlanesSynthetic"
+_CATEGORY_NUMBER = 6
+_ATTRIBUTES_GETTER: Dict[Tuple[int, ...], Callable[[str], Union[int, str, float]]] = {
+    (1, 3, 11, 13): float,
+    (14, 15): lambda x: x,
+    (12,): int,
+}
+
+
+def RarePlanesSynthetic(path: str) -> Dataset:
+    """`RarePlanesSynthetic <https://www.cosmiqworks.org/RarePlanes/>`_ dataset.
+
+    The file structure of RarePlanesSynthetic looks like::
+
+        <path>
+            images/
+                Atlanta_Airport_0_0_101_1837.png
+                ...
+            masks/
+                Atlanta_Airport_0_0_101_1837_mask.png
+                ...
+            xmls/
+                Atlanta_Airport_0_0_101_1837.xml
+                ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    category_ids = dataset.catalog.panoptic_mask.get_category_to_index()
+    segment = dataset.create_segment()
+    original_mask_dir = os.path.join(root_path, "masks")
+    new_mask_dir = os.path.join(root_path, "new_masks")
+    os.makedirs(new_mask_dir, exist_ok=True)
+    annotation_dir = os.path.join(root_path, "xmls")
+    for image_path in glob(os.path.join(root_path, "images", "*.png")):
+        segment.append(
+            _get_data(image_path, original_mask_dir, annotation_dir, new_mask_dir, category_ids)
+        )
+    return dataset
+
+
+def _get_data(
+    image_path: str,
+    original_mask_dir: str,
+    annotation_dir: str,
+    new_mask_dir: str,
+    category_ids: Dict[str, int],
+) -> Data:
+    stem = os.path.splitext(os.path.basename(image_path))[0]
+    mask_path = os.path.join(new_mask_dir, f"{stem}.png")
```

Comment:
`new_mask_path`?
Since using `new` and `original` to distinguish different masks in the code above, should the variable names below also follow this rule?